vision/object.go

```diff
@@ -17,19 +19,26 @@ type Object struct {
 
 // NewObject creates a new vision.Object from a point cloud with an empty label.
 func NewObject(cloud pc.PointCloud) (*Object, error) {
-	return NewObjectWithLabel(cloud, "")
+	return NewObjectWithLabel(cloud, "", nil)
 }
 
 // NewObjectWithLabel creates a new vision.Object from a point cloud with the given label.
-func NewObjectWithLabel(cloud pc.PointCloud, label string) (*Object, error) {
+func NewObjectWithLabel(cloud pc.PointCloud, label string, geometry *commonpb.Geometry) (*Object, error) {
 	if cloud == nil {
 		return NewEmptyObject(), nil
 	}
-	box, err := pc.BoundingBoxFromPointCloudWithLabel(cloud, label)
+	if geometry == nil {
+		box, err := pc.BoundingBoxFromPointCloudWithLabel(cloud, label)
+		if err != nil {
+			return nil, err
+		}
+		return &Object{cloud, box}, nil
+	}
+	geom, err := spatialmath.NewGeometryFromProto(geometry)
```

Comment:
What if a label is provided? Does NewGeometryFromProto take into account labels?

Reply:
Okay worked something in here such that if you actually provide a label it'll use that one. If you don't, it'll default to whatever label you have in the geometry proto object